components/private/Math/src/MDCMath.h

```diff
@@ -94,3 +94,20 @@ static inline CGFloat MDCSqrt(CGFloat value) {
   return sqrtf(value);
 #endif
 }
+
+/**
+ Expand `rect' to the smallest rect containing it with pixel-aligned origin and size.
+
+ @see CGRectIntegral
+ */
+static inline CGRect MDCRectAlignToScale(CGRect rect, CGFloat scale) {
+  if (MDCCGFloatEqual(scale, 1)) {
+    return CGRectIntegral(rect);
+  }
+  CGPoint newOrigin =
+      CGPointMake(MDCFloor(rect.origin.x * scale) / scale, MDCFloor(rect.origin.y * scale) / scale);
```

Comment:
This currently only works with standardized rects.
We likely want
CGFloat xOffset = MDCFloor(CGRectGetXOffset(rect) * scale) / scale;
CGFloat yOffset = MDCFloor(CGRectGetYOffset(rect) * scale) / scale;

There is some additional logic around widths.

Prefer CGRect accessor macros over rect.origin or rect.size unless you have explicitly called CGRectStandardize()